### PR TITLE
Fix #35 'kubectl logs -f' implemented

### PR DESCRIPTION
**Pull request by antoinetran**
# Summary

Implement `kubectl logs -f` in interlink VK and API side. In case of following parameter set, it will reads and write continuously with a buffer from API to VK.
Also improved InterLink internal log, so that a sessionContext (eg: `GetLogs#12345`) is created and added in HTTP header, so that we can also follow the HTTP request from VK to API to Plugin with the same id.
---

**Related issue :**
https://github.com/interTwin-eu/interlink-slurm-plugin/issues/35

### CODE REVIEW

File: pkg/virtualkubelet/execute.go [hunk 1]
```diff
@@ -107,6 +110,9 @@ func PingInterLink(ctx context.Context, config Config) (bool, int, error) {
 	defer spanHTTP.End()
 	defer types.SetDurationSpan(startHTTPCall, spanHTTP)
 
+	// Add session number for end-to-end from VK to API to InterLink plugin (eg interlink-slurm-plugin)
+	AddSessionContext(req, "PingInterLink#"+strconv.Itoa(rand.Intn(100000)))
```
Comment (dciangot) [inline pkg/virtualkubelet/execute.go:114]: @antoinetran can you expand/describe a bit what this stands for? Which is the problem to be solved here?

**Review by dciangot — COMMENTED** [review]
@antoinetran thank you for this huge contribution. It looks very good to me, just find one comment where I need a clarification on the design. Otherwise I think it's good to go

File: pkg/virtualkubelet/execute.go (hunk 1 above)
Comment (antoinetran) [reply pkg/virtualkubelet/execute.go:114]: Yes, this is, as the comment says, to add a HTTP session context information from end to end. When I started debugging in the test environment, I got many:
* getstatus call from InterLink VK, that goes to API, and then InterLink Slurm plugin
* a few kubectl logs -f from my test case
So I got so many logs in the interlink API and slurm that I couldn't know which logs is from which kubectl call. Adding this information allows one to know: first what API we are calling (GetStatus? GetLogs? CreatePod?) and which HTTP request it was at the beginning (with the random number like GetLogs#12345).

Comment (antoinetran) [reply pkg/virtualkubelet/execute.go:114]: You can see an example here https://github.com/interTwin-eu/interlink-slurm-plugin/pull/39#issuecomment-2514862524

**Comment by antoinetran** [comment]
I should have put some info in PullRequest instead of issue35:
* a few tests:
  * https://github.com/interTwin-eu/interlink-slurm-plugin/pull/39#issuecomment-2514862524
  * https://github.com/interTwin-eu/interlink-slurm-plugin/pull/39#issuecomment-2514845521

TODO for later: when doing Ctrl+C to kubectl logs -f, stops following logs for that session. But I don't know how to detect that and react.

**Comment by dciangot** [comment]
@antoinetran  looks like the last commit broke the compilation, can you take a look?

**Comment by dciangot** [comment]
mmm weird... repeating the tests locally shows no problem... maybe they broke on the force push. 

So, nvm. I'm push this through.

**Review by dciangot — APPROVED** [review]
LGTM

**Pull request merged**